### PR TITLE
Update Blobstore Servlets

### PR DESCRIPTION
**Pull request by saakshisaxena**
- Update UploadedImageInfoServlet to better handle exceptions.
- Refactor code in BlobstoreUploadUrlServlet.
- Delete unused Image class.
- Solve the problem of storing states about a single request in the UploadedImageInfoServlet by keeping an HttpSession variable.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/PhotoShoppingServlet.java [hunk 1]
```diff
@@ -38,11 +39,14 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // TODO: Based on request.getParameter("photo-category"), call methods from photo detection
-    // classes, passing request.getParameter("blob-key") as argument. These methods build the 
+    // TODO: Based on HttpSession session variable's "photoCategory", call methods from photo detection
+    // classes, passing session.getAttribute("blobKeyString") as argument. These methods build the
```
Comment (ralucatudor) [inline src/main/java/com/google/sps/servlets/PhotoShoppingServlet.java:43]:
```suggestion
    // classes, passing {@code session.getAttribute("blobKeyString")} as argument. These methods build the 
```

File: src/main/java/com/google/sps/servlets/PhotoShoppingServlet.java [hunk 2]
```diff
@@ -38,11 +39,14 @@
...
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // TODO: Based on request.getParameter("photo-category"), call methods from photo detection
-    // classes, passing request.getParameter("blob-key") as argument. These methods build the 
+    // TODO: Based on HttpSession session variable's "photoCategory", call methods from photo detection
+    // classes, passing session.getAttribute("blobKeyString") as argument. These methods build the 
     // shopping query and call the {@code query} method from GoogleShoppingQuerier.
 
-    String shoppingQuery = getQuery(request.getParameter("photo-category"));
+    // Get the session, which contains user-specific data
+	HttpSession session = request.getSession();
```
Comment (ralucatudor) [inline src/main/java/com/google/sps/servlets/PhotoShoppingServlet.java:47]: Nit: extra spaces before HttpSession

File: src/main/java/com/google/sps/servlets/UploadedImageInfoServlet.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    String blobKeyString = blobKey.getKeyString();
+
+    if (request.getParameter("photo-category").isEmpty()) {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST, "Client must select a photo category when submitting the form.");
+    }
+    String photoCategory = request.getParameter("photo-category");
+
+    // Store the item in the session
+	session.setAttribute("blobKeyString", blobKeyString);
```
Comment (ralucatudor) [inline src/main/java/com/google/sps/servlets/UploadedImageInfoServlet.java:61]: Nit: extra spaces before "session"

File: src/main/webapp/shopping-results.js [hunk 4]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
```
Comment (ralucatudor) [inline src/main/webapp/shopping-results.js]: Nit: extra line here

File: src/main/java/com/google/sps/servlets/UploadedImageInfoServlet.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    // User submitted form without selecting a file, so the BlobKey is empty. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    return blobKey;
+  }
+
```
Comment (ralucatudor) [inline src/main/java/com/google/sps/servlets/UploadedImageInfoServlet.java:93]: Nit: extra line here

File: src/main/java/com/google/sps/servlets/UploadedImageInfoServlet.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+/** Get and upload the HttpSession session variable with the image info */
+@WebServlet("/get-image-info")
+public class UploadedImageInfoServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
```
Comment (ralucatudor) [inline src/main/java/com/google/sps/servlets/UploadedImageInfoServlet.java:40]: Should we keep this extra line here or remove it? 🤔

**Review by ralucatudor — APPROVED** [review]

File: src/main/webapp/shopping-results.js [hunk 7]
```diff
@@ -21,10 +21,7 @@ async function buildShoppingResultsUI() {
   // search query results. The request returns a JSON with data about each product
   // from the Google Shopping results page.
 
-  // Build the URL to be fetched - add (hard-coded, for now) parameters to identify 
-  // the uploaded photo.
-  const fetchURL = 
-      `/photo-shopping-request?photo-category=product`;
+  const fetchURL = `/photo-shopping-request`;
```
Comment (ralucatudor) [inline src/main/webapp/shopping-results.js:24]: Also, now we can change the \` \` to ' '

**Review by ralucatudor — COMMENTED** [review]
I'm thinking that, considering this PR contains the updates that we made to the Blobstore servlets, the name _"Store different users' info separately"_ doesn't do justice to the PR content. How about renaming it to "Update Blobstore Servlets" and add in the description some things like:

- Update UploadedImageInfoServlet to better handle exceptions. 
- Also refactor code in BlobstoreUploadUrlServlet. 
- Delete unused Image class.
(These changes I've pushed, so I got the description from my previous PR)

- Solve the problem of storing states about a single request in the UploadedImageInfoServlet by keeping an HttpSession variable.

File: src/main/java/com/google/sps/servlets/UploadedImageInfoServlet.java (hunk 3 above)
Comment (saakshisaxena) [reply src/main/java/com/google/sps/servlets/UploadedImageInfoServlet.java:61]: It doesn't show any extra space in my editor idk why it shows it here

File: src/main/java/com/google/sps/servlets/UploadedImageInfoServlet.java (hunk 6 above)
Comment (saakshisaxena) [reply src/main/java/com/google/sps/servlets/UploadedImageInfoServlet.java:40]: I think it's better if we remove it considering that we didn't leave a new line after other method signatures.

**Pull request closed without merging**